rebar.config.script

```diff
@@ -1,74 +1,37 @@
-        %% remove trailing " and conver to string
-        binary_to_list(binary:part(Apps, 0, size(Apps) - 1));
-    (_) ->
-        io:format("WARNING! The configure.out file was not created, using minimal configuration~n"),
-        ""
-end,
-
-ReadFromConfig =
-fun(Path) ->
-        case filelib:is_file(Path) of
-            true ->
-                MaybeReadFromConfig(file:read_file(Path));
-            _ ->
-                Script = filename:join(["tools", "configure"]),
-                os:cmd(Script ++ " with-all without-jingle-sip"),
-                MaybeReadFromConfig(file:read_file("configure.out"))
-        end
-end,
-
-GetEnvApps =
-fun() ->
-        case os:getenv("APPS") of
-            false ->
-                ConfigurePath = "configure.out",
-                ReadFromConfig(ConfigurePath);
-            EnvApps ->
-                EnvApps
-        end
-end,
-
-DevAppsToInclude =
-fun() ->
-        case os:getenv("DEVNODE") of
-            false -> [];
-            _ -> [meck]
-        end
-end,
-
-FirstApps = [kernel],
-RequiredApps = [mongooseim, inets, tools, compiler],
-OptionalApps = [{nksip, [nkservice, nkpacket, nklib]}],
-
-EnvApps = GetEnvApps(),
-
-PrepareApp =
-fun(App) ->
-        case lists:keyfind(App, 1, OptionalApps) of
-            false ->
-                [App];
-            {App, DepsOfApp} ->
-                lists:map(fun(V) -> {V, load} end, [App | DepsOfApp])
-        end
-end,
-
-SetupIncludedApps =
-fun(Config, EnvApps) ->
-        RelxCfg = proplists:get_value(relx, Config),
-        {release, Desc, _Apps} = lists:keyfind(release, 1, RelxCfg),
-        EnvAppsToInclude = [ list_to_atom(App) || App <- string:tokens(EnvApps, " \n\r") ],
-        AppsToIncludeIn = lists:usort(RequiredApps ++ DevAppsToInclude() ++ EnvAppsToInclude),
-        Apps = lists:subtract(AppsToIncludeIn, FirstApps),
-        NewReleaseCfg = {release, Desc, FirstApps ++ lists:flatmap(PrepareApp, Apps)},
-        NewRelxCfg = lists:keyreplace(release, 1, RelxCfg, NewReleaseCfg),
-        lists:keyreplace(relx, 1, Config, {relx, NewRelxCfg})
-end,
-
-SetupIncludedApps(CONFIG, EnvApps).
+BaseApps = [kernel, stdlib, sasl, inets, tools, compiler, mongooseim,
+            {mnesia, load},
+            {cets, load},
+            {erlang_doctor, load},
+            {syslogger, load},
+            {cpool, load},
+            {backoff, load},
+            {mysql, load},
+            {eodbc, load},
+            {epgsql, load},
+            {eredis, load},
+            {cqerl, load},
+            {tirerl, load},
+            {erlcloud, load},
+            {amqp_client, load},
+            {exometer_core, load},
+            {exometer_report_graphite, load},
+            {exometer_report_statsd, load},
+            {prometheus, load},
+            {prometheus_httpd, load},
+            {prometheus_cowboy, load},
+            {nksip, load},
+            {nkservice, load},
+            {nkpacket, load},
+            {nklib, load}
```

Comment:
It's quite unexpected to me that we completely disregard the `APPS` list created by `configure` - e.g. if we configure without `odbc`, we still load `eodbc`. The whole effort of crafting the list of apps in `tools/configure` is now for nothing? Users would expect that it still has effect - maybe it's even still documented. I wouldn't like to repeat something that happened with shapers.

Reply:
Hmm... If we configure without `odbc` we still need to have `odbc` headers for compilation to pass, which is something that has bitten me many times. This logic is a bit simpler, and also more self-contained, for the possibility when we want to modify the list of apps to be loaded without starting. On the other hand, what could be interesting is when we don't want to even load an app 🤔

I think I can easily rework the case of `start|load|none`, let me give it a try.

Also, `configure` could take some modernisation to use https://www.erlang.org/doc/apps/stdlib/argparse.html too :)

Reply:
@chrzaszcz, of course, `relx` accepts the `none` mode, meaning code is just put in the path but not even loading happens. This makes startup time even faster :)

This still supersedes the `configure` script, so the question is how to reconcile these mechanisms. Replicating `start` vs `none` logic for more applications is a lot more convoluted in the `configure` script IMO, that's why I chose this mechanism as the best one. If that's ok for you I can remove that responsibility from the `configure` script and update docs accordingly.

Reply:
> If that's ok for you I can remove that responsibility from the `configure` script and update docs accordingly.

Yes, that's one of the ways to solve it. I just don't want to have contradictory mechanisms coexisting.